funnel/forms/space.py

```diff
@@ -92,6 +93,30 @@ class ParticipantForm(forms.Form):
         validators=[forms.validators.DataRequired(u"Select at least one event")])
 
 
+# TODO: Move to Baseframe
+class ValidFile(object):
+    def __init__(self, allowed_exts=[], message=None):
+        if not message:
+            message = __(u"Please upload a valid file.")
+        self.message = message
+        self.allowed_exts = allowed_exts
+
+    def __call__(self, form, field):
+        ext = os.path.splitext(field.data.filename)[1].strip(".")
```

Comment:
Have you tried this on Windows with IE? Filename is as reported by the browser based on the native OS, while `os.path` uses the conventions of the host OS. You will also get an internal server error if someone uploads a file without an extension.